scripts/index.js

```diff
@@ -52,7 +52,9 @@ const env = getClientEnvironment(process.env.PUBLIC_URL || ''); // eslint-disabl
 /**
  * We need to update the webpack dev config in order to remove the use of webpack devserver
  */
-config.entry = config.entry.filter(fileName => !fileName.match(/webpackHotDevClient/));
+if (major < 4) {
+  config.entry = config.entry.filter(fileName => !fileName.match(/webpackHotDevClient/));
+}
 config.plugins = config.plugins.filter(
   plugin => !(plugin instanceof webpack.HotModuleReplacementPlugin)
```

Comment:
Error: [React Refresh] Hot Module Replacement (HMR) is not enabled! React Refresh requires HMR to function properly.

I tried to use it in my game but gave me the error above you can find the code of the game here: https://github.com/keremciu/house-of-decks/blob/ws-implement/client/package.json

if I run `FAST_REFRESH=false cra-build-watch` this way it works but I'd like to learn how react-refresh-webpack plugin is affecting process

Reply:
how about wrapping this filter function with process.env.FAST_REFRESH?

Reply:
why do you need to use cra-build-watch in your project?

Reply:
Finally got the same error as you, updated the PR, I disabled the fast refresh because its not needed with this.

Reply:
I use it for development, basically I run `nodemon` on server then I needed to watch my changes on client side too.

I can confirm by disabling FAST_REFRESH, it works 💯

Reply:
Tested these changes in my app that uses cra-build-watch and these worked with react-scripts v4. :+1:

Reply:
Thanks for investigating this.